### PR TITLE
Migrate tutorials and a how-to guide to v2 primitives

### LINKED ISSUE #505: Use SamplerV2 and EstimatorV2 in documentation
Once support is added for `SamplerV2` (#488) and once Qiskit Aer supports `EstimatorV2` (https://github.com/Qiskit/qiskit-aer/issues/2078), we should update our uses of the primitives in the documentation to use the v2 version of the primitives.

### LINKED ISSUE #509: Update cutting tutorials to submit all jobs before calling `.result()` to obtain them
It'd be great to update the cutting tutorials to submit all jobs before calling `.result()` to obtain them, instead of doing both submission and retrieval in the same dict comprehension.  That way, all jobs will be submitted and they can wait in the queue at the same time instead of being submitted and retrieved serially.

### LINKED ISSUE #511: Use qiskit-ibm-runtime in local mode in cutting tutorials
Following https://docs.quantum.ibm.com/api/migration-guides/local-simulators and https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/pull/500#discussion_r1538186636, it'd be great to bring our cutting tutorials in line with using `Sampler` or `SamplerV2` from qiskit-ibm-runtime, together with either `AerSimulator` or (likely even better) a fake backend from qiskit-ibm-runtime.  Then, we will no longer need to use the `Sampler` from Qiskit Aer, and it will be more obvious how to use the code with Qiskit Runtime without having a special commented-out block in each tutorial.

One thing that needs to happen first is that https://github.com/Qiskit/qiskit-ibm-runtime/pull/1558 must make it into a release of qiskit-ibm-runtime.

### LINKED ISSUE #535: Use `Batch` mode instead of `Session` mode in cutting tutorials
`Session` is recommended for iterative workloads, but since we know all the circuits upfront, we should use `Batch` instead.  Let's also remove the unnecessary `as session` from the context managers, since we don't use it.

### PR DESCRIPTION
**Pull request by garrison**
This builds on #513 and follows https://docs.quantum.ibm.com/api/migration-guides/local-simulators.

I've added this to the 0.7.0 milestone, but it need not block the release because this portion can be moved to the 0.7.1 milestone and backported once it is ready.

- Fixes #511.
- Fixes #509.
- Fixes #505.
- Fixes #535.
- Progress toward #477.

#### Remaining action items

- [x] use sampler with fake backend instead of aer backend
- [x] update other tutorials, and any how-tos that need it
- [x] investigate whether the qiskit-ibm-runtime warning about using session in local mode is truly necessary: https://github.com/Qiskit/qiskit-ibm-runtime/issues/1626

### CODE REVIEW

**Comment by garrison** [comment]
With this change, the tutorials are a bit slower, but I think it is acceptable.  It's just so nice to use Qiskit Runtime with fake backends everywhere.  If a user wants to use a real backend, they need only modify the `backend`.

#### [Before](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/actions/runs/8728049081/job/23946937195)

```
============================= slowest 10 durations =============================
30.34s call     docs/circuit_cutting/tutorials/02_gate_cutting_to_reduce_circuit_depth.ipynb::
7.10s call     docs/circuit_cutting/tutorials/01_gate_cutting_to_reduce_circuit_width.ipynb::
7.03s call     docs/circuit_cutting/cutqc/tutorials/tutorial_1_automatic_cut_finding.ipynb::
6.43s call     docs/circuit_cutting/tutorials/03_wire_cutting_via_move_instruction.ipynb::
6.23s call     docs/circuit_cutting/cutqc/tutorials/tutorial_2_manual_cutting.ipynb::
5.32s call     docs/circuit_cutting/how-tos/how_to_specify_cut_wires.ipynb::
3.70s call     docs/circuit_cutting/tutorials/04_automatic_cut_finding.ipynb::
3.57s call     docs/circuit_cutting/how-tos/how_to_generate_exact_sampling_coefficients.ipynb::
1.46s call     docs/circuit_cutting/how-tos/how_to_generate_exact_quasi_dists_from_sampler.ipynb::
```

#### [After](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/actions/runs/8728791185/job/23949361137?pr=552)

```
============================= slowest 10 durations =============================
40.44s call     docs/circuit_cutting/tutorials/02_gate_cutting_to_reduce_circuit_depth.ipynb::
17.30s call     docs/circuit_cutting/tutorials/01_gate_cutting_to_reduce_circuit_width.ipynb::
16.22s call     docs/circuit_cutting/tutorials/03_wire_cutting_via_move_instruction.ipynb::
7.09s call     docs/circuit_cutting/cutqc/tutorials/tutorial_2_manual_cutting.ipynb::
7.01s call     docs/circuit_cutting/cutqc/tutorials/tutorial_1_automatic_cut_finding.ipynb::
6.93s call     docs/circuit_cutting/how-tos/how_to_specify_cut_wires.ipynb::
3.94s call     docs/circuit_cutting/tutorials/04_automatic_cut_finding.ipynb::
2.91s call     docs/circuit_cutting/how-tos/how_to_generate_exact_sampling_coefficients.ipynb::
1.34s call     docs/circuit_cutting/how-tos/how_to_generate_exact_quasi_dists_from_sampler.ipynb::
```

File: docs/circuit_cutting/tutorials/02_gate_cutting_to_reduce_circuit_depth.ipynb
Comment (garrison) [inline docs/circuit_cutting/tutorials/02_gate_cutting_to_reduce_circuit_depth.ipynb:1]: Just a note: there are much fewer single-qubit gates in `transpiled_qc` after moving to `FakeManilaV2`. The zoom level of that circuit ends up changing as a result, which can be noticed from the rich diff on github.

**Comment by caleb-johnson** [comment]
> With this change, the tutorials are a bit slower, but I think it is acceptable. It's just so nice to use Qiskit Runtime with fake backends everywhere. If a user wants to use a real backend, they need only modify the `backend`.
> 
> #### [Before](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/actions/runs/8728049081/job/23946937195)
> ```
> ============================= slowest 10 durations =============================
> 30.34s call     docs/circuit_cutting/tutorials/02_gate_cutting_to_reduce_circuit_depth.ipynb::
> 7.10s call     docs/circuit_cutting/tutorials/01_gate_cutting_to_reduce_circuit_width.ipynb::
> 7.03s call     docs/circuit_cutting/cutqc/tutorials/tutorial_1_automatic_cut_finding.ipynb::
> 6.43s call     docs/circuit_cutting/tutorials/03_wire_cutting_via_move_instruction.ipynb::
> 6.23s call     docs/circuit_cutting/cutqc/tutorials/tutorial_2_manual_cutting.ipynb::
> 5.32s call     docs/circuit_cutting/how-tos/how_to_specify_cut_wires.ipynb::
> 3.70s call     docs/circuit_cutting/tutorials/04_automatic_cut_finding.ipynb::
> 3.57s call     docs/circuit_cutting/how-tos/how_to_generate_exact_sampling_coefficients.ipynb::
> 1.46s call     docs/circuit_cutting/how-tos/how_to_generate_exact_quasi_dists_from_sampler.ipynb::
> ```
> 
> #### [After](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/actions/runs/8728791185/job/23949361137?pr=552)
> ```
> ============================= slowest 10 durations =============================
> 40.44s call     docs/circuit_cutting/tutorials/02_gate_cutting_to_reduce_circuit_depth.ipynb::
> 17.30s call     docs/circuit_cutting/tutorials/01_gate_cutting_to_reduce_circuit_width.ipynb::
> 16.22s call     docs/circuit_cutting/tutorials/03_wire_cutting_via_move_instruction.ipynb::
> 7.09s call     docs/circuit_cutting/cutqc/tutorials/tutorial_2_manual_cutting.ipynb::
> 7.01s call     docs/circuit_cutting/cutqc/tutorials/tutorial_1_automatic_cut_finding.ipynb::
> 6.93s call     docs/circuit_cutting/how-tos/how_to_specify_cut_wires.ipynb::
> 3.94s call     docs/circuit_cutting/tutorials/04_automatic_cut_finding.ipynb::
> 2.91s call     docs/circuit_cutting/how-tos/how_to_generate_exact_sampling_coefficients.ipynb::
> 1.34s call     docs/circuit_cutting/how-tos/how_to_generate_exact_quasi_dists_from_sampler.ipynb::
> ```

Agree with this. I really like the FakeBackends for demos for that same reason

**Comment by caleb-johnson** [comment]
> With the change to a fake backend, the first tutorial notebook now takes over 15 minutes to execute on my laptop. 😭
> 
> EDIT: I changed to a 5-qubit fake backend, and now it runs within a reasonable time.

I'm seeing the same behavior on my end. I wonder if we should run some quick numbers and report this up. Maybe this is expected, but it seems like a massive slowdown over V1

**Comment by garrison** [comment]
> I wonder if we should run some quick numbers and report this up.

It is all a bit opaque to me, but here is what I suspect is happening: When we use a 5-qubit fake backend, the code simulates all 5 qubits, even when only 2 or 4 are in use, which can be a couple times slower.  Plus, there might be some additional slowdown due to applying the noise model from the chosen fake backend.  So in that sense it may not be surprising that it is slower to use a fake backend than a noiseless `AerSimulator`.  Also, my hunch is consistent with [the massive slowdown I saw](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/pull/552#issuecomment-2059525545) when using the same code with `FakeTorino` (133 qubits).  On that many qubits, it's essentially forced to use the MPS simulator if it doesn't know just to ignore the unused qubits (and perhaps one cannot model the noise "correctly" without including unused qubits in the simulation -- I don't know).

**Review by caleb-johnson — APPROVED** [review]
This looks great, thanks for putting this together

I don't think the slowdown when trying to use `FakeBackend`s with more qubits is something we can control.

**Pull request merged**